packages/core/src/common/message-service-protocol.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable } from "inversify";
+
+export const messageServicePath = '/services/messageService';
+
+export enum MessageType {
+    Error = 1,
+    Warning = 2,
+    Info = 3,
+    Log = 4
+}
+
+export const MessageClient = Symbol('MessageClient');
+export interface MessageClient {
+    /**
+     * Show a message of the given type and possible actions to the user.
+     * Resolve to a chosen action.
+     * Never reject.
+     */
+    showMessage(type: MessageType, message: string, ...actions: string[]): Promise<string | undefined>;
```

Comment:
no titles, in the LSP they are of `MessageActionItem` type:
```ts
interface MessageActionItem {
	/**
	 * A short title like 'Retry', 'Open Log' etc.
	 */
	title: string;
}
```

Commands could be executed in `then` clause of the returned promise depending on the result:
```ts
const result = await messageService.info('Some Questiont`, 'Option 1', 'Option2')
if (result === 'Option 1') {
    this.commands.execute(myCommand);
}
```